CHANGELOG.md

```diff
@@ -6,6 +6,23 @@
 
 ### Compiler
 
+- Added more a verbose error when a pattern is attempted to be used after a spread.
+
+```
+error: Syntax error
+  ┌─ /src/parse/error.gleam:4:17
+  │
+4 │         [rest.., last] -> 1
+  │                  ^^^^ Invalid pattern
```

Comment:
Technically the highlighted bit _is_ a valid pattern. I think it would be nicer if the message was similar to the same you get for trying to append to a list with the spread operator.

When you try to use it to append it looks like this:
```
error: Syntax error
  ┌─ /src/main.gleam:4:10
  │
4 │   [1, ..[1, 2, 3], 3]
  │       ^^^^^^^^^^^ I wasn't expecting elements after this spread

A spread can only be used to prepend elements to lists like this: `[first, ..rest]`.
```

So here we might give something like this:
```
error: Syntax error
  ┌─ /src/main.gleam:4:10
  │
4 │   [1, ..xs, last]
  │       ^^^^ I wasn't expecting other patterns after this spread

A spread can only be used to match on the entire end of a list.
It is not possible to extract items from the end of a list using pattern
matching because that would require walking through the entire list.
```